### PR TITLE
Upstream: Clean up lockorder data of destroyed mutexes

### PR DESCRIPTION
**Pull request by bitcartel**
bitcoin/bitcoin#7846

### CODE REVIEW

**Comment by daira** [comment]
https://github.com/bitcoin/bitcoin/pull/8427 is a different PR. I think you mean https://github.com/bitcoin/bitcoin/pull/7846

File: src/sync.cpp [hunk 1]
```diff
@@ -96,25 +109,21 @@ static void push_lock(void* c, const CLockLocation& locklocation, bool fTry)
...
-            if (lockorders.count(p1))
+            if (lockdata.lockorders.count(p1))
                 continue;
-            lockorders[p1] = (*lockstack);
+            lockdata.lockorders[p1] = (*lockstack);
 
             std::pair<void*, void*> p2 = std::make_pair(c, i.first);
-            if (lockorders.count(p2)) {
-                potential_deadlock_detected(p1, lockorders[p2], lockorders[p1]);
-                break;
```
Comment (daira) [inline src/sync.cpp]: There's a difference between this and the upstream diff. Bitcoin adds the ``break`` statement in https://github.com/sipa/bitcoin/commit/660ff174f299a6dfb28f7725dfbb08f140a5474b#diff-ff1d9ab20c19185dbba33a95551fb6b2R91 , and Zcash adds it in https://github.com/bitcartel/zcash/commit/660ff174f299a6dfb28f7725dfbb08f140a5474b#diff-ff1d9ab20c19185dbba33a95551fb6b2R91 . But then in https://github.com/sipa/bitcoin/commit/9493803f4a60a234c2712bba000a6b9c93ffba94 , Bitcoin removes the ``break`` statement. We don't have the latter patch; do we need it (that is, does the upstream PR assume that it has been applied)?

**Comment by daira** [comment]
Not ACKing this yet because I don't know whether we need https://github.com/sipa/bitcoin/commit/9493803f4a60a234c2712bba000a6b9c93ffba94 .

**Comment by bitcartel** [comment]
@daira Move to 1.0.1 ?

**Comment by daira** [comment]
NACK due to undefined behaviour, as noted in https://medium.com/@g.andrew.stone/a-short-tour-of-bitcoin-core-4558744bf18b . Let's close this and do #1879 instead.

**Comment by nathan-at-least** [comment]
Closing in favor of #1879.

**Pull request closed without merging**